### PR TITLE
Convert logs to use structured logs

### PR DESCRIPTION
**Pull request by farah**
Part of #347

### CODE REVIEW

File: pkg/descheduler/strategies/lownodeutilization.go [hunk 1]
```diff
@@ -102,23 +102,23 @@ func LowNodeUtilization(ctx context.Context, client clientset.Interface, strateg
...
 	if len(lowNodes) == 0 {
-		klog.V(1).Infof("No node is underutilized, nothing to do here, you might tune your thresholds further")
+		klog.V(1).InfoS("No node is underutilized, nothing to do here, you might tune your thresholds further")
 		return
 	}
 	klog.V(1).InfoS("Total number of underutilized nodes", "totalNumber", len(lowNodes))
 
 	if len(lowNodes) < strategy.Params.NodeResourceUtilizationThresholds.NumberOfNodes {
-		klog.V(1).Infof("number of nodes underutilized (%v) is less than NumberOfNodes (%v), nothing to do here", len(lowNodes), strategy.Params.NodeResourceUtilizationThresholds.NumberOfNodes)
+		klog.V(1).InfoS("number of nodes underutilized is less than NumberOfNodes, nothing to do here", "underutilizedNodes", len(lowNodes), "numberOfNodes", strategy.Params.NodeResourceUtilizationThresholds.NumberOfNodes)
```
Comment (seanmalloy) [inline pkg/descheduler/strategies/lownodeutilization.go:111]:
```suggestion
		klog.V(1).InfoS("Number of nodes underutilized is less than NumberOfNodes, nothing to do here", "underutilizedNodes", len(lowNodes), "numberOfNodes", strategy.Params.NodeResourceUtilizationThresholds.NumberOfNodes)
```

File: pkg/descheduler/evictions/evictions.go [hunk 2]
```diff
@@ -227,7 +227,7 @@ func (ev *evictable) IsEvictable(pod *v1.Pod) bool {
 	}
 
 	if len(checkErrs) > 0 && !HaveEvictAnnotation(pod) {
-		klog.V(4).Infof("Pod %s in namespace %s is not evictable: Pod lacks an eviction annotation and fails the following checks: %v", pod.Name, pod.Namespace, errors.NewAggregate(checkErrs).Error())
+		klog.V(4).InfoS(errors.NewAggregate(checkErrs).Error(), "Pod lacks an eviction annotation and fails the following checks", "pod", klog.KObj(pod), "namespace", pod.Namespace)
```
Comment (seanmalloy) [inline pkg/descheduler/evictions/evictions.go:230]:
```suggestion
		klog.V(4).InfoS(errors.NewAggregate(checkErrs).Error(), "Pod lacks an eviction annotation and fails the following checks", "pod", klog.KObj(pod))
```

File: pkg/descheduler/strategies/lownodeutilization.go (hunk 1 above)
Comment (farah) [reply pkg/descheduler/strategies/lownodeutilization.go]: fixed

File: pkg/descheduler/evictions/evictions.go (hunk 2 above)
Comment (farah) [reply pkg/descheduler/evictions/evictions.go]: fixed

File: pkg/descheduler/strategies/lownodeutilization.go [hunk 3]
```diff
@@ -252,14 +252,14 @@ func evictPodsFromTargetNodes(
...
-		klog.V(2).Infof("AllPods:%v, nonRemovablePods:%v, removablePods:%v", len(node.allPods), len(nonRemovablePods), len(removablePods))
+		klog.V(2).InfoS("AllPods", len(node.allPods), "nonRemovablePods", len(nonRemovablePods), "removablePods", len(removablePods))
 
 		if len(removablePods) == 0 {
 			klog.V(1).InfoS("No removable pods on node, try next node", "node", klog.KObj(node.node))
 			continue
 		}
 
-		klog.V(1).Infof("evicting pods based on priority, if they have same priority, they'll be evicted based on QoS tiers")
+		klog.V(1).InfoS("evicting pods based on priority, if they have same priority, they'll be evicted based on QoS tiers")
```
Comment (seanmalloy) [inline pkg/descheduler/strategies/lownodeutilization.go:262]:
```suggestion
		klog.V(1).InfoS("Evicting pods based on priority, if they have same priority, they'll be evicted based on QoS tiers")
```

File: pkg/descheduler/evictions/evictions.go [hunk 4]
```diff
@@ -227,7 +227,7 @@ func (ev *evictable) IsEvictable(pod *v1.Pod) bool {
 	}
 
 	if len(checkErrs) > 0 && !HaveEvictAnnotation(pod) {
-		klog.V(4).Infof("Pod %s in namespace %s is not evictable: Pod lacks an eviction annotation and fails the following checks: %v", pod.Name, pod.Namespace, errors.NewAggregate(checkErrs).Error())
+		klog.V(4).InfoS(errors.NewAggregate(checkErrs).Error(), "Pod lacks an eviction annotation and fails the following checks", "pod", klog.KObj(pod))
```
Comment (seanmalloy) [inline pkg/descheduler/evictions/evictions.go:230]:
```suggestion
		klog.V(4).InfoS("Pod lacks an eviction annotation and fails the following checks", "pod", klog.KObj(pod), "checks", errors.NewAggregate(checkErrs).Error())
```

File: pkg/descheduler/strategies/lownodeutilization.go (hunk 3 above)
Comment (farah) [reply pkg/descheduler/strategies/lownodeutilization.go]: fixed

File: pkg/descheduler/evictions/evictions.go (hunk 4 above)
Comment (farah) [reply pkg/descheduler/evictions/evictions.go]: fixed

**Comment by seanmalloy** [comment]
@farah thanks for your help with converting to the new structured logging API.

/lgtm
/assign @damemi

**Review by damemi — APPROVED** [review]
/approve

**Pull request merged**